From f491b982c4eac4c493104d30ffaf031312d520dc Mon Sep 17 00:00:00 2001
From: lzc256 <i@lzc256.com>
Date: Sat, 20 Jun 2026 16:44:48 +0800
Subject: [PATCH] player.go: add friendsEnabled/acceptInvitesEnabled to
 UpdatePlayer

---
 player.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/player.go b/player.go
index d8b3569..57f50df 100644
--- a/player.go
+++ b/player.go
@@ -265,6 +265,8 @@ func (app *App) UpdatePlayer(
 	capeReader *io.Reader,
 	capeURL *string,
 	deleteCape bool,
+	friendsEnabled bool,
+	acceptInvitesEnabled bool,
 ) (Player, error) {
 	if caller == nil {
 		return Player{}, NewBadRequestUserError("Caller cannot be null.")
@@ -299,6 +301,9 @@ func (app *App) UpdatePlayer(
 		player.FallbackPlayer = *fallbackPlayer
 	}

+	player.FriendsEnabled = friendsEnabled
+	player.AcceptInvitesEnabled = acceptInvitesEnabled
+
 	if skinModel != nil {
 		if !IsValidSkinModel(*skinModel) {
 			return Player{}, NewBadRequestUserError("Invalid skin model.")

--
2.54.0
