diff --git a/src/worker/crawler.go b/src/worker/crawler.go
index b501ec3..cdd944b 100644
--- a/src/worker/crawler.go
+++ b/src/worker/crawler.go
@@ -200,7 +200,7 @@ func listItems(f model.Feed, db storage.Storage) ([]model.Item, error) {
 			LastRefreshed:    &now,
 		})
 	}
-	return ConvertItems(feed.Items, f), nil
+	return FilterItems(ConvertItems(feed.Items, f)), nil
 }
 
 func getCharset(res *http.Response) string {
