From f491b982c4eac4c493104d30ffaf031312d520dc Mon Sep 17 00:00:00 2001
From: lzc256 <i@lzc256.com>
Date: Sat, 20 Jun 2026 16:44:48 +0800
Subject: [PATCH] front.go: parse friendsEnabled/acceptInvitesEnabled form
 values

---
 front.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/front.go b/front.go
index 0308b00..acd9fac 100644
--- a/front.go
+++ b/front.go
@@ -1262,6 +1262,8 @@ func FrontUpdatePlayer(app *App) func(c *echo.Context) error {
 		deleteSkin := c.FormValue("deleteSkin") == "on"
 		capeURL := nilIfEmpty(c.FormValue("capeUrl"))
 		deleteCape := c.FormValue("deleteCape") == "on"
+		friendsEnabled := c.FormValue("friendsEnabled") == "on"
+		acceptInvitesEnabled := c.FormValue("acceptInvitesEnabled") == "on"

 		var player Player
 		result := app.DB.Preload("User").First(&player, "uuid = ?", playerUUID)
@@ -1309,6 +1311,8 @@ func FrontUpdatePlayer(app *App) func(c *echo.Context) error {
 			capeReader,
 			capeURL,
 			deleteCape,
+			friendsEnabled,
+			acceptInvitesEnabled,
 		)
 		if err != nil {
 			var userError *UserError

--
2.54.0
